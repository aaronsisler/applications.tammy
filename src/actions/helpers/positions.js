export const setPositions = (positions) => ({
    type: 'SET_POSITIONS',
    positions
});
