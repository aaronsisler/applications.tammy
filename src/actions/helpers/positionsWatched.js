export const addSubscription = (positionId, subscriptionLevel) => ({
    type: 'ADD_SUBSCRIPTION',
    positionId,
    subscriptionLevel
});

export const removeSubscription = (positionId) => ({
    type: 'REMOVE_SUBSCRIPTION',
    positionId
});

export const setPositionsWatched = (positionsWatched) => ({
    type: 'SET_POSITIONS_WATCHED',
    positionsWatched
});

export const setSubscriptionLevel = (positionId, subscriptionLevel) => ({
    type: 'SET_SUBSCRIPTION_LEVEL',
    positionId,
    subscriptionLevel
});
