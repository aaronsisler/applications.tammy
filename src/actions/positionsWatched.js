import database from 'Firebase/firebase';
import {
    addSubscription,
    removeSubscription,
    setPositionsWatched,
    setSubscriptionLevel
} from 'Actions/helpers/positionsWatched';

export const startAddSubscription = (positionId, subscriptionLevel) => (dispatch, getState) => {
    const { uid: userId } = getState().auth;
    return database.ref(`position_watch/${userId}/${positionId}`).update({ subscriptionLevel })
        .then(() => dispatch(addSubscription(positionId, subscriptionLevel)));
}

export const startRemoveSubscription = (positionId) => (dispatch, getState) => {
    const { uid: userId } = getState().auth;
    return database.ref(`position_watch/${userId}/${positionId}`).remove()
        .then(() => dispatch(removeSubscription(positionId)));
}

export const startSetPositionsWatched = () => (dispatch, getState) => {
    const { uid: userId } = getState().auth;

    return database.ref(`position_watch/${userId}`).once('value').then((snapshot) => {
        const positionsWatched = [];
        snapshot.forEach((childSnapshot) => {
            positionsWatched.push({
                positionId: childSnapshot.key,
                ...childSnapshot.val(),
            });
        });

        return dispatch(setPositionsWatched(positionsWatched));
    });
}

export const startSetSubscriptionLevel = (positionId, subscriptionLevel) => (dispatch, getState) => {
    const { uid: userId } = getState().auth;
    return database.ref(`position_watch/${userId}/${positionId}`).update({ subscriptionLevel })
        .then(() => dispatch(setSubscriptionLevel(positionId, subscriptionLevel)));
}



