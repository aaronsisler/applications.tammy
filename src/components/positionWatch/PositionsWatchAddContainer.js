import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import PositionsList from 'Shared/position/PositionsList';
import PositionWatchAddDetails from 'PositionWatch/PositionWatchAddDetails';

export class PositionsWatchAddContainer extends React.Component {
    constructor(props) {
        super(props);
    }

    handleMobileClassname = () => this.props.positionId ? "inbox_mobile" : ""

    render() {
        const { positions, positionsWatched } = this.props;
        const positionsConcat = [];
        positions.map((position) => {
            const positionFind = positionsWatched.find((positionWatched) => positionWatched.positionId === position.positionId);
            if (!positionFind) {
                positionsConcat.push({ ...position });
            }
        });

        return (
            <div className="inbox_container">
                {this.props.positionsWatched &&
                    <div className="inbox_widget">
                        <div className={`inbox_list ${this.handleMobileClassname()}`}>
                            <Link className="nav_link" to="/dashboard">Back to Dashboard</Link>
                            <PositionsList positions={positionsConcat} linkRoute={'position_watch_add'} />
                        </div>
                        <PositionWatchAddDetails {...this.props} />
                    </div>
                }
            </div>
        );
    }
}

/* istanbul ignore next */
const mapStateToProps = (state, props) => ({
    positionId: props.match.params.positionId,
    positions: state.positions,
    positionsWatched: state.positionsWatched,
});

export default connect(mapStateToProps)(PositionsWatchAddContainer);

PositionsWatchAddContainer.propTypes = {
    positionId: PropTypes.string,
    positions: PropTypes.array,
    positionsWatched: PropTypes.array,
};

