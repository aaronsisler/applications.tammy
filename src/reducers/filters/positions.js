const positionsFilterReducerDefaultState =
{
    text: '',
};

const positionsFilterReducer = (state = positionsFilterReducerDefaultState, action) => {
    switch (action.type) {
        case 'SET_POSITIONS_TEXT_FILTER':
            return {
                ...state,
                text: action.text
            };
        case 'CLEAR_POSITIONS_TEXT_FILTER':
            return {
                ...state,
                text: ''
            };
        default:
            return state;
    }
};

export default positionsFilterReducer;
