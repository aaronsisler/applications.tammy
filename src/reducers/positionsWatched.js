const positionsWatchedReducerDefaultState = [];

const positionsWatchedReducer = (state = positionsWatchedReducerDefaultState, action) => {
    switch (action.type) {
        case 'SET_POSITIONS_WATCHED':
            return action.positionsWatched;
        case 'ADD_SUBSCRIPTION':
                return [...state, action.positionWatched];
        case 'REMOVE_SUBSCRIPTION':
            return state.filter((positionwatched) => positionwatched.positionId != action.positionId);
        case 'SET_SUBSCRIPTION_LEVEL':
            return state.map((positionWatched) => {
                if (positionWatched.positionId === action.positionId) {
                    return {
                        ...positionWatched,
                        subscriptionLevel: action.subscriptionLevel
                    }
                }
                return positionWatched;
            })
        default:
            return state;
    }
};

export default positionsWatchedReducer;

