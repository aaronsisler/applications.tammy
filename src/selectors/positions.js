// Get filtered positions
export default (positions, { text = '' }) => positions
        .filter((position) => {
            const titleMatch = position.title.toLowerCase().includes(text.toLowerCase());
            const jobIdMatch = position.jobId.toLowerCase().includes(text.toLowerCase());
            return jobIdMatch || titleMatch;
        })
