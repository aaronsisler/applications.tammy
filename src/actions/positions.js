import database from '../firebase/firebase';
import { setPositions } from 'Actions/helpers/positions';

export const startSetPositions = () => (dispatch) =>
    database.ref(`positions`).once('value').then((snapshot) => {
        const positions = [];
        snapshot.forEach((childSnapshot) => {
            positions.push({
                positionId: childSnapshot.key,
                ...childSnapshot.val(),
            });
        });

        dispatch(setPositions(positions));
    });
