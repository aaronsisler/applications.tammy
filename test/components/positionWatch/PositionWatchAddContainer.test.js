import React from 'react';
import { shallow } from 'enzyme';
import { PositionsWatchAddContainer } from 'PositionWatch/PositionsWatchAddContainer';
import positionsWatched from '../../fixtures/positionsWatched';
import positions, { positionId } from '../../fixtures/positions';

describe('PositionsWatchAddContainer', () => {
    let wrapper;

    const buildWrapper = (positionsWatchedInput = [], positionIdInput) => {
        wrapper = shallow(
            <PositionsWatchAddContainer
                positionId={positionIdInput}
                positions={positions}
                positionsWatched={positionsWatchedInput}
            />
        );
    };

    describe('when positionId is available', () => {
        describe('when all positions are watched', () => {
            it('should render PositionsWatchAddContainer correctly', () => {
                buildWrapper(positionsWatched, positionId);

                expect(wrapper).toMatchSnapshot();
            });
        });

        describe('when some positions are watched', () => {
            it('should render PositionsWatchAddContainer correctly', () => {
                const [positionWatched] = positionsWatched;

                buildWrapper([positionWatched], positionId);

                expect(wrapper).toMatchSnapshot();
            });
        });

        describe('when NO positions are watched', () => {
            it('should render PositionsWatchAddContainer correctly', () => {
                buildWrapper(undefined, positionId);

                expect(wrapper).toMatchSnapshot();
            });
        });
    });

    describe('when positionId is NOT available', () => {
        it('should render PositionsWatchAddContainer correctly', () => {
            buildWrapper(positionsWatched, undefined);

            expect(wrapper).toMatchSnapshot();
        });
    });
});
