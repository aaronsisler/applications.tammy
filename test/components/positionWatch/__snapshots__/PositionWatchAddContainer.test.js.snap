// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PositionsWatchAddContainer when positionId is NOT available should render PositionsWatchAddContainer correctly 1`] = `
<div
  className="inbox_container"
>
  <div
    className="inbox_widget"
  >
    <div
      className="inbox_list "
    >
      <Link
        className="nav_link"
        replace={false}
        to="/dashboard"
      >
        Back to Dashboard
      </Link>
      <PositionsList
        linkRoute="position_watch_add"
        positions={Array []}
      />
    </div>
    <Connect(PositionWatchAddDetails)
      positions={
        Array [
          Object {
            "content": Object {
              "description": "first description",
              "requirements": Array [
                "Skill 1",
                "Skill 2",
              ],
              "responsibilities": Array [
                "Responsibility 1",
                "Responsibility 2",
              ],
            },
            "jobId": "first job id",
            "location": "first location",
            "positionId": "firstPositionId",
            "title": "first title",
          },
          Object {
            "content": Object {
              "description": "second description",
              "requirements": Array [
                "Skill 1",
                "Skill 2",
              ],
              "responsibilities": Array [
                "Responsibility 1",
                "Responsibility 2",
              ],
            },
            "jobId": "second job id",
            "location": "second location",
            "positionId": "secondPositionId",
            "title": "second title",
          },
        ]
      }
      positionsWatched={
        Array [
          Object {
            "content": Object {
              "description": "first description",
              "requirements": Array [
                "Skill 1",
                "Skill 2",
              ],
              "responsibilities": Array [
                "Responsibility 1",
                "Responsibility 2",
              ],
            },
            "jobId": "first job id",
            "location": "first location",
            "notificationLevel": "ALL",
            "positionId": "firstPositionId",
            "title": "first title",
          },
          Object {
            "content": Object {
              "description": "second description",
              "requirements": Array [
                "Skill 1",
                "Skill 2",
              ],
              "responsibilities": Array [
                "Responsibility 1",
                "Responsibility 2",
              ],
            },
            "jobId": "second job id",
            "location": "second location",
            "notificationLevel": "REQUIRED",
            "positionId": "secondPositionId",
            "title": "second title",
          },
        ]
      }
    />
  </div>
</div>
`;

exports[`PositionsWatchAddContainer when positionId is available when NO positions are watched should render PositionsWatchAddContainer correctly 1`] = `
<div
  className="inbox_container"
>
  <div
    className="inbox_widget"
  >
    <div
      className="inbox_list inbox_mobile"
    >
      <Link
        className="nav_link"
        replace={false}
        to="/dashboard"
      >
        Back to Dashboard
      </Link>
      <PositionsList
        linkRoute="position_watch_add"
        positions={
          Array [
            Object {
              "content": Object {
                "description": "first description",
                "requirements": Array [
                  "Skill 1",
                  "Skill 2",
                ],
                "responsibilities": Array [
                  "Responsibility 1",
                  "Responsibility 2",
                ],
              },
              "jobId": "first job id",
              "location": "first location",
              "positionId": "firstPositionId",
              "title": "first title",
            },
            Object {
              "content": Object {
                "description": "second description",
                "requirements": Array [
                  "Skill 1",
                  "Skill 2",
                ],
                "responsibilities": Array [
                  "Responsibility 1",
                  "Responsibility 2",
                ],
              },
              "jobId": "second job id",
              "location": "second location",
              "positionId": "secondPositionId",
              "title": "second title",
            },
          ]
        }
      />
    </div>
    <Connect(PositionWatchAddDetails)
      positionId="firstPositionId"
      positions={
        Array [
          Object {
            "content": Object {
              "description": "first description",
              "requirements": Array [
                "Skill 1",
                "Skill 2",
              ],
              "responsibilities": Array [
                "Responsibility 1",
                "Responsibility 2",
              ],
            },
            "jobId": "first job id",
            "location": "first location",
            "positionId": "firstPositionId",
            "title": "first title",
          },
          Object {
            "content": Object {
              "description": "second description",
              "requirements": Array [
                "Skill 1",
                "Skill 2",
              ],
              "responsibilities": Array [
                "Responsibility 1",
                "Responsibility 2",
              ],
            },
            "jobId": "second job id",
            "location": "second location",
            "positionId": "secondPositionId",
            "title": "second title",
          },
        ]
      }
      positionsWatched={Array []}
    />
  </div>
</div>
`;

exports[`PositionsWatchAddContainer when positionId is available when all positions are watched should render PositionsWatchAddContainer correctly 1`] = `
<div
  className="inbox_container"
>
  <div
    className="inbox_widget"
  >
    <div
      className="inbox_list inbox_mobile"
    >
      <Link
        className="nav_link"
        replace={false}
        to="/dashboard"
      >
        Back to Dashboard
      </Link>
      <PositionsList
        linkRoute="position_watch_add"
        positions={Array []}
      />
    </div>
    <Connect(PositionWatchAddDetails)
      positionId="firstPositionId"
      positions={
        Array [
          Object {
            "content": Object {
              "description": "first description",
              "requirements": Array [
                "Skill 1",
                "Skill 2",
              ],
              "responsibilities": Array [
                "Responsibility 1",
                "Responsibility 2",
              ],
            },
            "jobId": "first job id",
            "location": "first location",
            "positionId": "firstPositionId",
            "title": "first title",
          },
          Object {
            "content": Object {
              "description": "second description",
              "requirements": Array [
                "Skill 1",
                "Skill 2",
              ],
              "responsibilities": Array [
                "Responsibility 1",
                "Responsibility 2",
              ],
            },
            "jobId": "second job id",
            "location": "second location",
            "positionId": "secondPositionId",
            "title": "second title",
          },
        ]
      }
      positionsWatched={
        Array [
          Object {
            "content": Object {
              "description": "first description",
              "requirements": Array [
                "Skill 1",
                "Skill 2",
              ],
              "responsibilities": Array [
                "Responsibility 1",
                "Responsibility 2",
              ],
            },
            "jobId": "first job id",
            "location": "first location",
            "notificationLevel": "ALL",
            "positionId": "firstPositionId",
            "title": "first title",
          },
          Object {
            "content": Object {
              "description": "second description",
              "requirements": Array [
                "Skill 1",
                "Skill 2",
              ],
              "responsibilities": Array [
                "Responsibility 1",
                "Responsibility 2",
              ],
            },
            "jobId": "second job id",
            "location": "second location",
            "notificationLevel": "REQUIRED",
            "positionId": "secondPositionId",
            "title": "second title",
          },
        ]
      }
    />
  </div>
</div>
`;

exports[`PositionsWatchAddContainer when positionId is available when some positions are watched should render PositionsWatchAddContainer correctly 1`] = `
<div
  className="inbox_container"
>
  <div
    className="inbox_widget"
  >
    <div
      className="inbox_list inbox_mobile"
    >
      <Link
        className="nav_link"
        replace={false}
        to="/dashboard"
      >
        Back to Dashboard
      </Link>
      <PositionsList
        linkRoute="position_watch_add"
        positions={
          Array [
            Object {
              "content": Object {
                "description": "second description",
                "requirements": Array [
                  "Skill 1",
                  "Skill 2",
                ],
                "responsibilities": Array [
                  "Responsibility 1",
                  "Responsibility 2",
                ],
              },
              "jobId": "second job id",
              "location": "second location",
              "positionId": "secondPositionId",
              "title": "second title",
            },
          ]
        }
      />
    </div>
    <Connect(PositionWatchAddDetails)
      positionId="firstPositionId"
      positions={
        Array [
          Object {
            "content": Object {
              "description": "first description",
              "requirements": Array [
                "Skill 1",
                "Skill 2",
              ],
              "responsibilities": Array [
                "Responsibility 1",
                "Responsibility 2",
              ],
            },
            "jobId": "first job id",
            "location": "first location",
            "positionId": "firstPositionId",
            "title": "first title",
          },
          Object {
            "content": Object {
              "description": "second description",
              "requirements": Array [
                "Skill 1",
                "Skill 2",
              ],
              "responsibilities": Array [
                "Responsibility 1",
                "Responsibility 2",
              ],
            },
            "jobId": "second job id",
            "location": "second location",
            "positionId": "secondPositionId",
            "title": "second title",
          },
        ]
      }
      positionsWatched={
        Array [
          Object {
            "content": Object {
              "description": "first description",
              "requirements": Array [
                "Skill 1",
                "Skill 2",
              ],
              "responsibilities": Array [
                "Responsibility 1",
                "Responsibility 2",
              ],
            },
            "jobId": "first job id",
            "location": "first location",
            "notificationLevel": "ALL",
            "positionId": "firstPositionId",
            "title": "first title",
          },
        ]
      }
    />
  </div>
</div>
`;
